src/sys/socket/mod.rs

```diff
@@ -2148,10 +2148,22 @@ pub fn bind(fd: RawFd, addr: &dyn SockaddrLike) -> Result<()> {
 /// Accept a connection on a socket
 ///
 /// [Further reading](https://pubs.opengroup.org/onlinepubs/9699919799/functions/accept.html)
-pub fn accept(sockfd: RawFd) -> Result<RawFd> {
-    let res = unsafe { libc::accept(sockfd, ptr::null_mut(), ptr::null_mut()) };
+pub fn accept<T: SockaddrLike>(sockfd: RawFd) -> Result<(RawFd, T)> {
+    let mut addr = mem::MaybeUninit::<T>::uninit();
+    let mut len = T::size();
 
-    Errno::result(res)
+    let res = unsafe {
+        libc::accept(sockfd, addr.as_mut_ptr() as *mut sockaddr, &mut len)
+    };
+
+    Errno::result(res)?;
+
+    let addr = unsafe { T::from_raw(addr.assume_init().as_ptr(), Some(len)) };
```

Comment:
I think this is UB since it's not guaranteed that all bytes of `addr` are initialized. A simple example would be if someone passed in a `SockaddrStorage` for an inet socket. The number of bytes written by the syscall would be less than the size of `SockaddrStorage`, so `assume_init()` would create a `SockaddrStorage` from a partially-initialized `SockaddrStorage`.